src/Engines/Extensions/engine.occlusionQuery.ts

```diff
@@ -271,6 +296,47 @@ Engine.prototype.endTimeQuery = function(token: _TimeToken): int {
     return -1;
 };
 
+Engine.prototype._captureGPUFrameTime = false;
+Engine.prototype._gpuFrameTime = new PerfCounter();
+
+Engine.prototype.getGPUFrameTimeCounter = function () {
+    return this._gpuFrameTime;
+};
+
+Engine.prototype.captureGPUFrameTime = function (value: boolean) {
+    if (value === this._captureGPUFrameTime) {
+        return;
+    }
+
+    this._captureGPUFrameTime = value;
+
+    if (value) {
+        this._onBeginFrameObserver = this.onBeginFrameObservable.add(() => {
+            if (!this._gpuFrameTimeToken) {
+                this._gpuFrameTimeToken = this.startTimeQuery();

```

Comment:
does it work when using occlusion query? I was under the impression that only one query was available at time

Reply:
Not occlusion query but timer query. But it was already there before :) In this file, there were occlusion + time query functions

Reply:
oh right..I'm a dick with file names

can you rename the file name to something simply like engine.query ?